sdk/search/azure-search-documents/tests/test_service_live.py

```diff
@@ -423,3 +580,40 @@ def test_create_or_update_datasource(self, api_key, endpoint, index_name, **kwar
         result = client.get_datasource("sample-datasource")
         assert result.name == "sample-datasource"
         assert result.description == "updated"
+
+    @SearchResourceGroupPreparer(random_name_enabled=True)
+    @SearchServicePreparer(schema=SCHEMA, index_batch=BATCH)
+    def test_create_or_update_datasource_if_unchanged(self, api_key, endpoint, index_name, **kwargs):
+        client = SearchServiceClient(endpoint, AzureKeyCredential(api_key)).get_datasources_client()
+        data_source = self._create_datasource()
+        created = client.create_datasource(data_source)
+        etag = created.e_tag
+
+        # Now update the data source
+        data_source.description = "updated"
+        client.create_or_update_datasource(data_source)
+
+        # prepare data source
+        data_source.e_tag = etag # reset to the original datasource
+        data_source.description = "changed"
+        with pytest.raises(HttpResponseError):
+            client.create_or_update_datasource(data_source, match_condition=MatchConditions.IfNotModified)
+            assert len(client.get_datasources()) == 1
```

Comment:
I think this assert is never evaluated (same below) since it would happen after `create_or_update_datasource` raises?

Reply:
right - will remove it